pkg/controller/cluster/main_controller.go

```diff
@@ -365,7 +373,7 @@ func (c *Controller) syncHandler(key string) error {
 	ss, err := c.statefulSetLister.StatefulSets(mi.Namespace).Get(mi.Name)
 	// If the resource doesn't exist, we'll create it
 	if apierrors.IsNotFound(err) {
-		ss = statefulsets.NewForCluster(mi, svc.Name)
+		ss = statefulsets.NewForCluster(mi, hlSvc.Name)
 		ss, err = c.kubeClientSet.AppsV1().StatefulSets(mi.Namespace).Create(ctx, ss, cOpts)
```

Comment:
I thin we are not handling this errors, i don't see any defer at the beginning of the function